pyomo/core/plugins/transform/scaling.py

```diff
@@ -0,0 +1,255 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and 
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain 
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from pyomo.core.base import Var, Constraint, Objective, _ConstraintData, _ObjectiveData, Suffix, value
+from pyomo.core.plugins.transform.hierarchy import Transformation
+from pyomo.core.kernel.component_map import ComponentMap
+from pyomo.util.components import rename_components
+from pyomo.core.base import TransformationFactory
+from pyomo.core.expr.current import replace_expressions
+
+
+@TransformationFactory.register('core.scale_model',
+                                doc="Scale model variables, constraints, and objectives.")
+class ScaleModel(Transformation):
+    """
+    Transformation to scale a model.
+
+    This plugin performs variable, constraint, and objective scaling on
+    a model based on the scaling factors in the suffix 'scaling_parameter'
+    set for the variables, constraints, and/or objective. This is typically 
+    done to scale the problem for improved numerical properties.
+
+    Supported transformation methods:
+        * :py:meth:`apply_to <pyomo.core.plugins.transform.scaling.ScaleModel.apply_to>`
+        * :py:meth:`create_using <pyomo.core.plugins.transform.scaling.ScaleModel.create_using>`
+        * :py:meth:`propagate_solution <pyomo.core.plugins.transform.scaling.ScaleModel.propagate_solution>`
+
+
+    Examples:
```

Comment:
Could you slightly modify the formatting of this docstring example so that it gets picked up by the doctests? See the example here: https://github.com/Pyomo/pyomo/blob/8cbe45bf0143f4d4793a55a8de8c901cb4528e3b/pyomo/core/base/reference.py#L284